packages/features/schedules/lib/use-schedule/useTimesForSchedule.ts

```diff
@@ -1,24 +1,110 @@
+      !isNaN(monthAfterExtraDaysFromDate) &&
+      dateMonth !== monthAfterExtraDaysFromDate) ||
+    (bookerLayout.layout === BookerLayouts.COLUMN_VIEW &&
+      !isNaN(dateMonth) &&
+      !isNaN(monthAfterColumnViewExtraDaysFromDate) &&
+      dateMonth !== monthAfterColumnViewExtraDaysFromDate) ||
+    ((bookerLayout.layout === BookerLayouts.MONTH_VIEW || bookerLayout.layout === "mobile") &&
+      (!isValidDate || isSameMonth) &&
+      isAfter2Weeks);
+
+  // FIXME: There is an issue with this logic
+  // If monthAfter1MonthFromDate = 5 and monthAfterColumnViewExtraDaysFromDate = 4, it will prefetch 2 months that is in total 3 months. In such a situation infact we just need to prefetch just one month and thus total of 2 months.
+  // So, we should make it something like monthAfterColumnViewExtraDaysFromDate > monthAfter1MonthFromDate, but this condition considering that extra days remains a few days only, won't be practically true and thus it looks like we really don't need this condition at all
+  // Practically, this conditon doesn't frequently arise because it needs to have date query param set. So, to be able to replicate this issue, one needs to select a date and then refresh the page
+  // Fix it in a follow up PR - There is a failing test that is skipped because of this issue
+  const doesExtraDaysNeedOneMoreMonth =
+    !isNaN(monthAfter1MonthFromDate) &&
+    !isNaN(monthAfterColumnViewExtraDaysFromDate) &&
+    // monthAfterColumnViewExtraDaysFromDate > monthAfter1MonthFromDate &&
+    monthAfterColumnViewExtraDaysFromDate != monthAfter1MonthFromDate;
+
+  // TODO: We need to refactor the approach to prefetch and prefetching must be done as a second non-blocking request instead of fetching all data together
+  const monthsToPrefetch: 1 | 2 =
+    // Don't know why we need to specifically prevent prefetching for non-Week view layouts when booker state isn't selecting_time(e.g. loading)
+    // This ends up causing different time ranges for different booker states and thus multiple getSchedule calls are made
+    // Steps to replicate: Simply go to http://acme.cal.local:3000/owner1/30min?overlayCalendar=true&layout=month_view&month=2025-09&date=2025-09-24. Two calls will be made, first one that prefetches 1 month and second one that prefetches 2 months.
+    ((bookerLayout.layout !== BookerLayouts.WEEK_VIEW && bookerState === "selecting_time") ||
+      bookerLayout.layout === BookerLayouts.COLUMN_VIEW) &&
+    doesExtraDaysNeedOneMoreMonth
+      ? 2
+      : 1;
+
+  return {
+    pretchFutureMonths,
+    monthsToPrefetch,
+  };
+};
+
 export const useTimesForSchedule = ({
   month,
-  monthCount,
   selectedDate,
   dayCount,
-  prefetchNextMonth,
+  bookerLayout,
 }: UseTimesForScheduleProps): [string, string] => {
+  // Get values from booker store
+  const [monthFromStore, bookerState] = useBookerStoreContext((state) => [state.month, state.state], shallow);
+
+  const resolvedMonth = monthFromStore ?? month ?? null;
```

Comment:
This is how the BookerWebWrapper and BookerPlatformWrapper were resolving month preferring the monthFromStore and falling back to monthFromProperty